rule/deep_exit.go

```diff
@@ -76,5 +77,6 @@ func (w *lintDeepExit) Visit(node ast.Node) ast.Visitor {
 func (w *lintDeepExit) mustIgnore(fd *ast.FuncDecl) bool {
 	fn := fd.Name.Name
 
-	return fn == "init" || fn == "main" || (w.isTestFile && fn == "TestMain")
+	return fn == "init" || fn == "main" || (w.isTestFile && fn == "TestMain") ||
+		(w.isTestFile && strings.HasPrefix(fn, "Example"))
```

Comment:
This might not be enough to validate it's an example function.

The signature is supposed to be a func() {}

I had a counter example with a function with a method named something like 

```go
func ExamplesProvider(t *testing.T) []string {
  // ...
}
```

There are also a lot of Example_Whatever(), Examples(), Example(), Example_Foo_Bar(), Example_foo()

https://github.com/search?q=language%3Ago+%22func+Example%22+path%3A*_test.go+&type=code

The knowledge about testable examples is not spread.

I think I found a code in govet or gocritic about validating tested examples. You could take a look.

Reply:
Thank you, fixed.